libbeat/cmd/instance/beat.go

```diff
@@ -285,6 +286,226 @@ func NewBeat(name, indexPrefix, v string, elasticLicensed bool, initFuncs []func
+	}
+
+	fqdnLookupCtx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	fqdn, err := h.FQDNWithContext(fqdnLookupCtx)
+	if err != nil {
+		// FQDN lookup is "best effort".  We log the error, fallback to
+		// the OS-reported hostname, and move on.
+		logp.Warn("unable to lookup FQDN: %s, using hostname = %s as FQDN", err.Error(), b.Info.Hostname)
+		b.Info.FQDN = b.Info.Hostname
+	} else {
+		b.Info.FQDN = fqdn
+	}
+
+	// initialize config manager
+	m, err := management.NewManager(b.Config.Management, b.Registry)
+	if err != nil {
+		return nil, fmt.Errorf("error creating new manager: %w", err)
+	}
+	b.Manager = m
+
+	if b.Manager.AgentInfo().Version != "" {
+		// During the manager initialization the client to connect to the agent is
+		// also initialized. That makes the beat to read information sent by the
+		// agent, which includes the AgentInfo with the agent's package version.
+		// Components running under agent should report the agent's package version
+		// as their own version.
+		// In order to do so b.Info.Version needs to be set to the version the agent
+		// sent. As this Beat instance is initialized much before the package
+		// version is received, it's overridden here. So far it's early enough for
+		// the whole beat to report the right version.
+		b.Info.Version = b.Manager.AgentInfo().Version
+		version.SetPackageVersion(b.Info.Version)
+	}
+
+	// build the user-agent string to be used by the outputs
+	b.GenerateUserAgent()
+
+	if err := b.Manager.CheckRawConfig(b.RawConfig); err != nil {
+		return nil, fmt.Errorf("error checking raw config: %w", err)
+	}
+
+	b.Beat.BeatConfig, err = b.BeatConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error setting BeatConfig: %w", err)
+	}
+
+	imFactory := settings.IndexManagement
+	if imFactory == nil {
+		imFactory = idxmgmt.MakeDefaultSupport(settings.ILM)
+	}
+	b.IdxSupporter, err = imFactory(nil, b.Beat.Info, b.RawConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error setting index supporter: %w", err)
+	}
+
+	processingFactory := settings.Processing
+	if processingFactory == nil {
+		processingFactory = processing.MakeDefaultBeatSupport(true)
+	}
+
+	b.processors, err = processingFactory(b.Info, logp.L().Named("processors"), b.RawConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error creating processors: %w", err)
+	}
+
+	reg := monitoring.Default.GetRegistry(b.Info.Name)
+	if reg == nil {
+		reg = monitoring.Default.NewRegistry(b.Info.Name)
+	}
+
+	// This should be replaced with static config for otel consumer
+	// but need to figure out if we want the Queue settings from here.
+	outputEnabled := b.Config.Output.IsSet() && b.Config.Output.Config().Enabled()
+	if !outputEnabled {
+		if b.Manager.Enabled() {
+			logp.Info("Output is configured through Central Management")
+		} else {
+			msg := "no outputs are defined, please define one under the output section"
+			logp.Info(msg)
```

Comment:
Should this log be at level error? Not configuring an output is an configuration error. Did you use info because the original code also uses info?

Reply:
Yeah, that was a copy of what the original code did.  I changed it to just return an error.  The caller of NewBeatReceiver can log the error correctly.